src/main/java/com/example/store/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,68 @@
+package com.example.store.controller;
+
+import com.example.store.dto.cartitem.CartItemDto;
+import com.example.store.dto.cartitem.CartItemRequestDto;
+import com.example.store.dto.cartitem.CartItemUpdateRequestDto;
+import com.example.store.dto.shoppingcart.ShoppingCartDto;
+import com.example.store.service.CartItemService;
+import com.example.store.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints to managing shopping cart")
+@RestController
+@RequestMapping(value = "/cart")
+@RequiredArgsConstructor
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final CartItemService cartItemService;
+
+    @GetMapping
+    @PreAuthorize("hasAuthority('USER')")
+    @Operation(summary = "Get all categories",
+            description = "Get a page of all available categories and can use sort")
+    public ShoppingCartDto getAll(Authentication authentication, Pageable pageable) {
+        return shoppingCartService.findShoppingCart(authentication.getName());
```

Comment:
imho it's more usual to work with Ids

Reply:
agree, using id is much better and faster